src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -295,7 +247,7 @@ private void WatchForFileSystemEventsThreadStart()
                 else if ((DoesPathPassNameFilter(eventPaths[i])) && ((_filterFlags & eventFlags[i]) != 0))
                 {
                     // Remove the base directory prefix (including trailing / that OS X adds)
-                    string relativePath = eventPaths[i].Remove(0, _fullDirectory.Length + 1);
+                    string relativePath = (eventPaths[i].Equals(_fullDirectory, StringComparison.CurrentCultureIgnoreCase) ? _fullDirectory : eventPaths[i].Remove(0, _fullDirectory.Length).TrimStart('/'));
```

Comment:
I think you want to be doing an ordinal comparison here (and I guess ignoring case, if that's what we do other places in OSX land).